### ISSUE TITLE
Weird formatting after update to stable version

### ISSUE THREAD

**Issue by ops1ops** [prettier/prettier#13477]
**Prettier 2.7.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEBtKc4AmAIgJZoAOANgIYCeRcAZtQK6UwAS1ATgZAXACSkKABpMaODABy+YmSp0GzNpx58IA4dAC6mALyYWkgMoxq8ADwAjCBEpxqUAHwAKADpRMmZpUmeASgBuTxEMHDxCUgoaeiZWdm0oMws4cUkZOWjFOJVEkRT4PUNjOEK4GzsHJzdPbwBCATyYJMEAW2oAczgAVW5KQKCQURAIchgSdGRQHm4IAHcABR4ENGQQakp5ujWR625qMABrKRNqNrgAGRI8ZF9JPYPj0-JDm867zYeQODbrQgEBEuTk6LC6cAAYhBuB0YBMoB8UKwYBBhiAABYwNqUADq6JI8AohzKqwJJAAbgTaOswGhdiAbpJuDBFgdOh1Pn40iAAFZoAAeJneDgAiiwIPBOd9XtwmesYLRyHA0GBuCRxmjyGrYDiSAQYOjkABGAAMJpGWogkhxB3I6y1yrg3HJcDRAEdxfBWWM1ki0ABaSKAtHcOAekih1ldDlIe7cyRtEjIGDcFjx4VwMUS12xr7c8zWXX6w1IABMIxT1BIlHeAGEIG0Yz80ABWNGlAAq1GsvrjI3JacEUAEsBMqvVMAAgsOzLQHFK4ABfRdAA)

```sh
--parser typescript
--print-width 100
--no-bracket-spacing
```

**Input:**

```tsx
const [needDisplayDefaultHardcodeIcon, setNeedDisplayDefaultHardcodeIcon] = useState<boolean>(
  false
);
const [needDisplayDefaultIconState, setNeedDisplayDefaultIconState] = useState<boolean>(
  !defaultIconImageUrl
);
```

**Output:**

```tsx
const [needDisplayDefaultHardcodeIcon, setNeedDisplayDefaultHardcodeIcon] =
  useState<boolean>(false);
const [needDisplayDefaultIconState, setNeedDisplayDefaultIconState] = useState<boolean>(
  !defaultIconImageUrl
);

```

**Expected behavior:**
In my opinion first useState should be formatted as second, because it has enough space, or maybe im missing something in configuration

**Comment by thorn0** [comment]
> In my opinion first useState should be formatted as second

Why not vice versa? Short single call arguments like `false` look bad (unbalanced) when they're printed on a separate line. In the logic that prints assignments, there is this heuristic that tries to prevent the "lone single argument on a separate line" situation. Probably it should be adjusted for arguments with `!`.

**Comment by ops1ops** [comment]
> > In my opinion first useState should be formatted as second
> 
> Why not vice versa? Short single call arguments like `false` look bad (unbalanced) when they're printed on a separate line. In the logic that prints assignments, there is this heuristic that tries to prevent the "lone single argument on a separate line" situation. Probably it should be adjusted for arguments with `!`.

Makes sense. The only thing that confuses me is that the current behavior is inconsistent. Either first useState should be formatted as second or vice versa
